pkg/common/dashboard.go

```diff
@@ -0,0 +1,136 @@
+package common
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/cnrancher/autok3s/pkg/settings"
+	k3dutil "github.com/k3d-io/k3d/v5/cmd/util"
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const HelmDashboardCommand = "helm-dashboard"
+
+func SwitchDashboard(ctx context.Context, enabled string) error {
+	if enabled == "true" {
+		// check if cluster list is empty
+		clusters, err := DefaultDB.ListCluster("")
+		if err != nil {
+			return err
+		}
+		if len(clusters) == 0 {
+			return errors.New("cannot enable helm-dashboard with empty cluster list")
+		}
+	}
+	if err := CheckCommandExist(HelmDashboardCommand); err != nil {
+		return err
+	}
+
+	// command execution validate
+	if err := checkDashboardCmd(); err != nil {
+		return err
+	}
+
+	isEnabled := settings.HelmDashboardEnabled.Get()
+	if !strings.EqualFold(isEnabled, enabled) {
+		if err := settings.HelmDashboardEnabled.Set(enabled); err != nil {
+			return err
+		}
+		if enabled == "true" {
+			dashboardPort := settings.HelmDashboardPort.Get()
+			if dashboardPort == "" {
+				freePort, err := k3dutil.GetFreePort()
```

Comment:
Is it better to use a static port instead of random one? By setting default value of `HelmDashboardPort` setting.

Reply:
I think it's easy to use from UI to enable helm-dashboard plugin by one click and UI will shows the dashboard link for each cluster, users don't need to keep in mind for the url. The port will also keep the same after disabled. 
If using CLI, we can set dashboard port by `--port` parameter.